lib/src/embedded/compiler/message-transformer.ts

```diff
@@ -46,177 +35,87 @@ export class MessageTransformer {
 
   // Decodes a buffer into an OutboundMessage.
   private decode(buffer: Buffer) {
-    let message;
     try {
-      message = toMessage(buffer);
+      const message = toMessage(buffer);
+      this.readInternal$.next(message);
     } catch (error) {
-      this.protocolErrorInternal$.next(error);
+      this.errorInternal$.next(error);
```

Comment:
Should this close upon encountering an error? Given that, should the error stream just be a promise instead?

Reply:
Yes, eventually—I'm planning to have the handler who instantiates the message transformer also handle closing it.

Reply:
I do think it's advantageous to keep it as an Observable for as long as possible, until we're ready to resolve and return its value to the API call